addons/pos_loyalty/models/loyalty_reward.py

```diff
@@ -11,3 +12,20 @@ def _get_discount_product_values(self):
         for vals in res:
             vals.update({'taxes_id': False})
         return res
+
+    def _should_compute_all_discount_product_pos(self, reward):
+        reward_discount_product_domain = reward._get_discount_product_domain()
+        if not reward_discount_product_domain:
+            return False
+        pos_product_product_loaded_fields = set(
+            self.env['pos.session'].POS_PRODUCT_PRODUCT_FIELDS) | {'id'}
+        for leaf in reward_discount_product_domain:
+            if len(leaf) == 1 and leaf in DOMAIN_OPERATORS:
```

Comment:
Is it necessary to check DOMAIN_OPERATORS? You can verify if it's an array or tuple with a length of 3.